tools/const_finder/lib/const_finder.dart

```diff
@@ -64,12 +59,13 @@ class _ConstVisitor extends RecursiveVisitor<void> {
 
   @override
   void visitConstructorInvocation(ConstructorInvocation node) {
-    final Class parentClass = node.target.parent as Class;
+    final Class parentClass = node.target.parent! as Class;
```

Comment:
Nit: Why `!` is necessary?

Reply:
`parent` is `TreeNode? parent`. I'm casting it to a non-null `Class`.

Reply:
AFAIK it is allowed to cast a value of any type (including nullable) to any type (including non-nullable), so `!` seems to be redundant (non-nullability of parent is checked by `as Class` cast anyway).

Reply:
ahh I'm following https://dart-lang.github.io/linter/lints/cast_nullable_to_non_nullable.html